cmd/api/src/api/v2/auth/auth_test.go

```diff
@@ -3137,3 +3168,404 @@ func TestActivateMFA_Success(t *testing.T) {
+					"message": "an internal error has occurred that is preventing the service from servicing this request"
+				}
+			]
+		}`
+
+		responseBodyWithDefaultTimestamp, err := replaceFieldValueInJsonString(rec.Body.String(), "timestamp", "0001-01-01T00:00:00Z")
+		require.NoError(tc, err)
+		assert.JSONEq(tc, expectedResponse, responseBodyWithDefaultTimestamp)
+	})
+
+	t.Run("Success with data wrapper", func(tc *testing.T) {
+		mockCtrl := gomock.NewController(tc)
+		defer mockCtrl.Finish()
+
+		resources, mockDB := apitest.NewAuthManagementResource(mockCtrl)
+		router := createRouter(endpointUrl, resources.CreateAuthToken, "POST")
+
+		admin, adminContext := createAdminUser(tc)
+
+		payload := map[string]any{
+			"user_id":    admin.ID.String(),
+			"token_name": "test token",
+		}
+
+		mockDB.EXPECT().GetUser(gomock.Any(), admin.ID).Return(admin, nil)
+
+		expectedToken := model.AuthToken{
+			UserID:     uuid.NullUUID{UUID: admin.ID, Valid: true},
+			Name:       null.StringFrom("test token"),
+			HmacMethod: "hmac-sha2-256",
+			Key:        "test-key",
+		}
+
+		mockDB.EXPECT().CreateAuthToken(gomock.Any(), gomock.Any()).Return(expectedToken, nil)
+
+		b, err := json.Marshal(payload)
+		require.NoError(tc, err)
+
+		req, err := http.NewRequestWithContext(adminContext, "POST", endpointUrl, bytes.NewReader(b))
+		require.NoError(tc, err)
+
+		req.Header.Set(headers.ContentType.String(), mediatypes.ApplicationJson.String())
+
+		rec := httptest.NewRecorder()
+		router.ServeHTTP(rec, req)
+
+		assert.Equal(tc, http.StatusOK, rec.Code)
+
+		var responseWrapper struct {
+			Data model.AuthToken `json:"data"`
+		}
+
+		err = json.Unmarshal(rec.Body.Bytes(), &responseWrapper)
+		require.NoError(tc, err)
+		assert.Equal(tc, expectedToken.Name.String, responseWrapper.Data.Name.String)
+	})
+}
+
+// Matrix testing pattern example for future reference
+func TestManagementResource_CreateAuthToken_Matrix(t *testing.T) {
+	type TestData struct {
+		testName             string
+		expectedResponseBody string
+		expectedResponseCode int
+		setupMocks           func(*testing.T, mocks.MockDatabase, *http.Request)
+		payload              map[string]any
+		userContext          func(*testing.T) (model.User, context.Context)
+	}
+
+	testData := []TestData{
+		{
+			testName:             "Request invalid",
+			expectedResponseBody: `{"http_status":400,"timestamp":"0001-01-01T00:00:00Z","request_id":"","errors":[{"context":"","message":"error unmarshalling JSON payload"}]}`,
+			expectedResponseCode: http.StatusBadRequest,
+			payload:              nil, // Will be handled specially in the test
+			userContext: func(t *testing.T) (model.User, context.Context) {
+				admin, adminCtx := createAdminUser(t)
+				return admin, adminCtx
+			},
+			setupMocks: func(t *testing.T, db mocks.MockDatabase, req *http.Request) {
+			},
+		},
+		// Add additional test cases here
+	}
+
+	for _, testArgs := range testData {
+		t.Run(testArgs.testName, func(tc *testing.T) {
```

Comment:
We should always use `t` for the `*testing.T` variable by convention. This will help us avoid accessing the wrong testing struct by accident and make tests more consistent

Reply:
This is still a concern